### PR TITLE
Amun inspection result analysis functions

### PR DESCRIPTION
**Pull request by shruthi-raghuraman**
- Added thoth lab functions from summer Amun analysis notebooks for statistical analysis

### CODE REVIEW

File: thoth/lab/inspection.py [hunk 1]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
+            b = p.get_bbox()
+            val = "{0:g}".format(b.x1 - b.x0)
+            if (val != '0'):
+                g.annotate(val, ((b.x1) + x_offset, b.y1 + y_offset))
+        count += 1
+
+
+def plot_distribution_of_jobs_combined_categories(df: pd.DataFrame, df_duration: pd.DataFrame,
+                                                  df_analyze: pd.DataFrame):
+    """Create trace stacked plot scaled by total jobs of each parameter within clusters if any.
```
Comment (CermakM) [inline thoth/lab/inspection.py]: Maybe a copy/paste typo?

File: thoth/lab/inspection.py [hunk 2]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
+
+    :param df: data frame with duration information as returned by process_inspection_results
+    :param df_categories:  filtered dataframe with columns to analyze as returned by columns_to_analyze
```
Comment (CermakM) [inline thoth/lab/inspection.py]: There is no such parameter

File: thoth/lab/inspection.py [hunk 3]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
+    df_final = pd.concat([i for i in lst_processed], axis=1)
+    return df_final
+
+
+def summary_trace_plot(df: pd.DataFrame, df_categories: pd.DataFrame, lst=[]):
+    """Create trace plot scaled by percentage of compositions of each parameter analyzed.
+
+    :param df: data frame with duration information as returned by process_inspection_results
+    :param df_categories: filtered dataframe with columns to analyze as returned by columns_to_analyze
+    :param lst: list of set or subset dataframes of data with the last value in the list being entire data set
```
Comment (CermakM) [inline thoth/lab/inspection.py]: This parameter is extremely confusing to me. Perhaps a type hint with `typings` would help at least a bit?

File: thoth/lab/inspection.py [hunk 4]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
+    return df.drop(list_of_unhashable_none_values, axis=1)
+
+
+# Function takes in dataframe
+def unique_value_count_by_feature_class(df: pd.DataFrame):
+    """Print unique count values per feature/class.
+
+    Print results per feature/class that are subdivided in subclasses that map to it.
+
+
```
Comment (CermakM) [inline thoth/lab/inspection.py]: Too many blank lines

File: thoth/lab/inspection.py [hunk 5]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
+            except TypeError:
+                lst_new = (list(df[i].values))
+                value_count = len([i for n, i in enumerate(lst_new) if i not in lst_new[n + 1:]])
+                if ((value_count >= low) and (value_count <= high)):
+                    print(i, value_count)
+                    lst_columns_to_analyze.append(i)
+                pass
+
+    # Filters data frame to columns with distinct value counts within the limit
+
```
Comment (CermakM) [inline thoth/lab/inspection.py]: Extra line

File: thoth/lab/inspection.py [hunk 6]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
     y_label = px.set_ylabel(y_label)
+
+
+def columns_to_analyze(df: pd.DataFrame, low=0, high=len(df_original),
+                       display_clusters=False, cluster_by_hue=False) -> pd.DataFrame:
+    """Print all columns within dataframe and count of unique column values within limit.
+
+    :param df: data frame to analyze as returned by `process_inspection_results'
+    :param low: the lower limit (0 if not specified) of distinct value counts
+    :param high: the upper limit (size of data set if not specified) of distinct value counts
```
Comment (CermakM) [inline thoth/lab/inspection.py]: Missing parameters

File: thoth/lab/inspection.py [hunk 7]
```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
...
+    :param colum: job duration/build duration columns from 'df'
+    """
+    df = df.reset_index()
+    for i in df_cat:
+        g = sns.FacetGrid(df, hue=i, margin_titles=True, height=5, aspect=1)
+        g.map(sns.regplot, column, "index", fit_reg=False, x_jitter=.1,  scatter_kws=dict(alpha=0.5))
+        g.add_legend()
+
+
+def concatenated_df(lst_of_df, column):
```
Comment (CermakM) [inline thoth/lab/inspection.py]: Missing type hints

**Comment by goern** [comment]
What's the status on this?

**Comment by pacospace** [comment]
> What's the status on this?

@shruthi-raghuraman

**Comment by goern** [comment]
any updates? @shruthi-raghuraman

**Comment by goern** [comment]
@shruthi-raghuraman ping

**Comment by shruthi-raghuraman** [comment]
Have opened new pr #165 (https://github.com/thoth-station/lab/pull/165) since I am unable to access this branch on my current laptop.

**Pull request closed without merging**